src/HearThis/UI/AndroidSyncDialog.cs

```diff
@@ -203,10 +203,64 @@ public MyMessageArgs(byte[] newData)
 		private void _syncButton_Click(object sender, EventArgs e)
 		{
 			AndroidIpAddress = _ipAddressBox.Text;
+			if (AndroidIpAddress.Contains("?"))
+			{
+				MessageBox.Show(
+					"You need to replace the three question marks in the box to the left with the number shown on the device in order to sync manually",
+					"Sorry");
+				return;
+			}
+
+			if (!ValidateIpAddress())
+			{
+				MessageBox.Show(
+					"The value in the address box does not appear to be a valid device address",
+					"Sorry");
+				return;
+			}
+
+			if (!IsPlausibleIpAddress())
+			{
+				if (MessageBox.Show(
+					"The device address you entered appears to be on a different local network. This usually won't work. Do you want to try anyway?",
+					"Warning", MessageBoxButtons.YesNo) == DialogResult.No)
+				{
+					return;
+				}
+			}
 			m_listener.StopListener();
 			HandleGotIpAddress();
 		}
 
+		private bool ValidateIpAddress()
+		{
+			var parts = AndroidIpAddress.Split('.');
+			if (parts.Length != 4)
```

Comment:
Technically, an IP address can be valid with fewer than 4 parts. I assume in HearThis' case, it will always have all 4?